http.go

```diff
@@ -92,7 +103,7 @@ func (client *Client) Request(rawPayload []byte, method, route string, query url
 				return nil, nil, err
 			}
 		}
-		resp, err := (&http.Client{}).Do(callSummary.HTTPRequest)
+		resp, err := DefaultHTTPClient.Do(callSummary.HTTPRequest)
```

Comment:
As well, using a `DefaultHTTPClient` allows us to only instantiate one `http.Client`, which we can do because it is thread-safe. If we want to prevent people setting it to nil, I can instead change it to a non-exported variable and we'll still have the benefits without the potential for users to get the package in a non-working state.